classes/local/store/s3/client.php

```diff
@@ -79,44 +79,91 @@ public function __wakeup() {
         // it will be serialised, so re-retrive them now.
         $config = manager::get_objectfs_config();
         $this->set_client($config);
-        $this->client->registerStreamWrapper();
+        if ($this->is_functional()) {
+            $this->client->registerStreamWrapper();
+        }
     }
 
-    public function set_client($config) {
-        $options = array(
-            'region' => $config->s3_region,
-            'version' => AWS_API_VERSION
-        );
+    /**
+     * Check if the client is functional.
+     * @return bool
+     */
+    private function is_functional() {
+        return !is_null($this->client);
+    }
 
-        if (empty($config->s3_usesdkcreds)) {
-            $options['credentials'] = array('key' => $config->s3_key, 'secret' => $config->s3_secret);
+    /**
+     * Check if the client configured properly.
+     *
+     * @param \stdClass $config Client config.
+     * @return bool
+     */
+    private function is_configured($config) {
+        if (empty($config->s3_bucket)) {
+            return false;
         }
 
-        if ($config->useproxy) {
-            $options['http'] = array('proxy' => $this->get_proxy_string());
+        if (empty($config->s3_region)) {
+            return false;
         }
 
-        // Support base_url config for aws api compatible endpoints.
-        if ($config->s3_base_url) {
-            $options['endpoint'] = $config->s3_base_url;
+        if (empty($config->s3_usesdkcreds)) {
+            if (empty($config->s3_key) && empty($config->s3_secret)) {
+                return false;
+            }
         }
```

Comment:
I think this was meant to be something like this according to https://github.com/catalyst/moodle-tool_objectfs/issues/467

```suggestion
        if (empty($config->s3_usesdkcreds)) {
            return false;
        }
        
        if (empty($config->s3_key) && empty($config->s3_secret)) {
            return false;
        }
```

Reply:
we want to check s3_key and s3_secret ONLY if s3_usesdkcreds is emptyt, don't we?

Reply:
Ah right. I get what you mean. I haven't pulled it back up but assuming `s3_usesdkcreds` is on the same lines as "Use the environment to determine the keys available" (to a certain extent), I think maybe this is the preferred form - i.e. don't nest the conditions, just add to it since it's all AND checks.

```suggestion
        if (empty($config->s3_usesdkcreds)
            && empty($config->s3_key) 
            && empty($config->s3_secret)) {
            return false;
        }
```

Reply:
I wrote it in a way a think about it as I wasn't sure if there could be any other combinations in case if s3_usesdkcreds is set. As there is only one at the moment, I'm changing to a suggested code.